Dev 2

Nokafan opened the pull request.

Commit:
Eternal cycling

Commit:
Simple solution

OleksiiTromsa commented on src/main/java/core/basesyntax/controller/ConsoleHandler.java:
```diff
@@ -0,0 +1,65 @@
+                    newUser.setPassword(password);
+                    addBets(scanner, newUser);
+                    UserDao newUserDao = new UserDaoImpl();
+                    newUserDao.addNewUser(newUser);
```
Lets move work with UserDao out of try block.
https://www.youtube.com/watch?v=pWpDdXGjNOE&feature=youtu.be
In this video Bogdan explains why should we do it.

OleksiiTromsa commented on src/main/java/core/basesyntax/controller/ConsoleHandler.java:
```diff
@@ -0,0 +1,65 @@
+                newBet.setValue(betValue);
+                newBet.setRisk(riskValue);
+                BetDao betDao = new BetDaoImpl();
+                betDao.add(newBet);
```
Same as UserDao

OleksiiTromsa commented on src/main/java/core/basesyntax/controller/ConsoleHandler.java:
```diff
@@ -0,0 +1,65 @@
+                newBet.setRisk(riskValue);
+                BetDao betDao = new BetDaoImpl();
+                betDao.add(newBet);
+            } catch (Exception e) {
```
In the same video Bogdan suggest to catch not Exception but NumberFormatException.
But if we do so, we can have case when somebody input such string as "Bob" and we will get  ArrayIndexOutOfBoundsException because we will have only 1 element in dataStringsBet. 
So if we will catch NumberFormatException instead of Exception we also should catch ArrayIndexOutOfBoundsException.

KseniiaMakarova replied:
I agree, better catch these two exceptions in one catch block

Nokafan replied:
Changed to NumberFormatException | IndexOutOfBoundsException e

OleksiiTromsa commented on src/main/java/core/basesyntax/storage/ListStorageUser.java:
```diff
@@ -0,0 +1,14 @@
+public class ListStorageUser {
+    public static List<User> userList = new ArrayList<>();
+
+    public void addUser(User user) {
+        userList.add(user);
+        System.out.println(user + " User added.");
+    }
```
I think class ListStorageUser has only one function - to store data. It shouldn't do something with it. We have UserDao for it. And you have already implemented method addNewUser() in UserDao. So I suggest to delete addUser from ListStorageUser.

Nokafan replied:
Ok, for this situation. But in fruit shop I was told to move some logic to store, so I used that expirience.

OleksiiTromsa commented on src/main/java/core/basesyntax/storage/ListStorageUser.java:
```diff
@@ -0,0 +1,14 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class ListStorageUser {
```
Suggestion: change ListStorageUser to UserStorage

OleksiiTromsa commented on src/main/java/core/basesyntax/storage/ListStorageBets.java:
```diff
@@ -0,0 +1,9 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class ListStorageBets {
```
Suggestion: change ListStorageBets to BetStorage

Nokafan replied:
It is plane list it is better to see what it is, storage could be db for example.

OleksiiTromsa commented on src/main/java/core/basesyntax/dao/UserDaoImpl.java:
```diff
@@ -0,0 +1,12 @@
+public class UserDaoImpl implements UserDao {
+    @Override
+    public void addNewUser(User user) {
+        ListStorageUser listStorageUser = new ListStorageUser();
+        listStorageUser.addUser(user);
```
Dao should work with database. And in your database you have:
public static List userList
It is static and you can call it by the class name: ListStorageUser.userList.add(user)
So it is no need to create new ListStorageUser.

Nokafan replied:
ok

OleksiiTromsa commented on src/main/java/core/basesyntax/model/Bet.java:
```diff
@@ -0,0 +1,69 @@
+    private int value;
+    private double risk;
+
+    public Bet() {
+    }
```
I think we do not need constructor without parameters for Bet. Because it has no sense to create Bet without value and risk. And if we give such possibility to another developers they can make mistakes in future :)

Nokafan replied:
But I have ID so I could activate this somewhere right before inputting in db. But ok fixed.

KseniiaMakarova commented on src/main/java/core/basesyntax/storage/ListStorageBets.java:
```diff
@@ -0,0 +1,9 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class ListStorageBets {
```
let's have one storage with two lists inside

KseniiaMakarova commented on src/main/java/core/basesyntax/controller/ConsoleHandler.java:
```diff
@@ -0,0 +1,70 @@
+                int id = user.getUserId();
+                int betValue = Integer.parseInt(dataStringsBet[0]);
+                double riskValue = Double.parseDouble(dataStringsBet[1]);
+                newBet = new Bet(user.getUserId(), betValue, riskValue);
```
```suggestion
                newBet = new Bet(id, betValue, riskValue);
```

Nokafan replied:
ops :)

KseniiaMakarova commented on src/main/java/core/basesyntax/controller/ConsoleHandler.java:
```diff
@@ -0,0 +1,70 @@
+        }
+    }
+
+    public void addBets(Scanner scanner, User user) {
```
```suggestion
    private void addBets(Scanner scanner, User user) {
```

Nokafan replied:
Fixed

KseniiaMakarova commented on src/main/java/core/basesyntax/storage/ListStorage.java:
```diff
@@ -0,0 +1,11 @@
+import java.util.List;
+
+public class ListStorage {
+    public static List<User> userList = new ArrayList<>();
+    public static List<Bet> betListStorage = new ArrayList<>();
```
make the names consistent

KseniiaMakarova commented on src/main/java/core/basesyntax/controller/ConsoleHandler.java:
```diff
@@ -0,0 +1,70 @@
+                    String password = dataStringsUser[2];
+                    newUser = new User(name, login, password);
+                    addBets(scanner, newUser);
+                } catch (NumberFormatException | IndexOutOfBoundsException e) {
```
do we need to catch NumberFormatException here?

Nokafan replied:
Nope :( copy paste from addBets method

KseniiaMakarova commented on src/main/java/core/basesyntax/controller/ConsoleHandler.java:
```diff
@@ -0,0 +1,70 @@
+import java.util.Scanner;
+
+public class ConsoleHandler {
+    UserDao newUserDao = new UserDaoImpl();
```
```suggestion
    UserDao userDao = new UserDaoImpl();
```

Nokafan replied:
Fixed

Nokafan committed:
Removed extra empty lines

Fixed commits

Fixed commits

Changed names of store, added Objects.requireNonNull

Removed Objects.requireNonNull

Removed one empty line
